### PR TITLE
PC_COLOR: use cameras to publish RGB pointcloud

### PR DESCRIPTION
**Pull request by RustyBamboo**

### CODE REVIEW

File: perception/mil_vision/src/pc_colorizer.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        }
+        else
+        {
+          point.b = 255;
+          point.g = 255;
+          point.r = 255;
+        }
+      }
+    }
+    temp_cloud->header.frame_id = "velodyne";
```
Comment (kev-the-dev) [inline perception/mil_vision/src/pc_colorizer.cpp]: Should be in whatever frame pointcloud is in

File: perception/mil_vision/src/pc_colorizer.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+  tf2_ros::Buffer tf_buffer_;
+  tf2_ros::TransformListener tf_listener_;
+
+public:
+  //! Initialize tf buffer and resolve camera feeds
+  Colorizer() : nh_(), tf_listener_(tf_buffer_)
+  {
+    std::vector<std::string> camera_names = { "/stereo/right/image_rect_color", "/camera/front/right/"
+                                                                                "image_rect_color" };
```
Comment (kev-the-dev) [inline perception/mil_vision/src/pc_colorizer.cpp]: What kind of wizardry is this? Didn't know this is valid syntax

File: perception/mil_vision/src/pc_colorizer.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    nh_.param<std::vector<std::string>>("/pc_colorizer/cameras", camera_names, camera_names);
+
+    // what topic to subscribe to
+    std::string pointcloud_subscriber = "/velodyne_points";
+    nh_.param<std::string>("/pc_colorizer/pointcloud_subscriber", pointcloud_subscriber, pointcloud_subscriber);
```
Comment (kev-the-dev) [inline perception/mil_vision/src/pc_colorizer.cpp]: Should always use [relative/private parameters](http://wiki.ros.org/roscpp/Overview/Parameter%20Server)

File: perception/mil_vision/src/pc_colorizer.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr temp_cloud(new pcl::PointCloud<pcl::PointXYZRGB>(*point_cloud));
+
+    // Remember transformation from pointcloud to camera frame
+    std::vector<Eigen::Affine3d> tf_cameras(cameras_.size());
+    auto tf_cameras_it = tf_cameras.begin();
+    for (auto &&cam : cameras_)
+    {
+      if (!tf_buffer_.canTransform((*cam)[0]->getCameraModelPtr()->tfFrame(), point_cloud->header.frame_id,
+                                   ros::Time(0)))
+        return;
```
Comment (kev-the-dev) [inline perception/mil_vision/src/pc_colorizer.cpp]: Warn/log error

File: perception/mil_vision/src/pc_colorizer.cpp (hunk 2 above)
Comment (RustyBamboo) [reply perception/mil_vision/src/pc_colorizer.cpp]: c++11 list initialization is a beautiful thing

Comment (kev-the-dev) [reply perception/mil_vision/src/pc_colorizer.cpp]: yeah yeah yeah but look closely at it... there's 3 strings

Comment (kev-the-dev) [reply perception/mil_vision/src/pc_colorizer.cpp]: Also just realized it should probaby default to /camera/front/right/image_rect_color and /camera/front/left/image_rect_color

**Review by kev-the-dev — APPROVED** [review]

**Comment by kev-the-dev** [comment]
Oh this doesn't build actually... check yo self

File: perception/mil_vision/src/pc_colorizer.cpp (hunk 2 above)
Comment (RustyBamboo) [reply perception/mil_vision/src/pc_colorizer.cpp]: I think this is an unintended result in clang format. Might be a bug. Dunno. I added an extra camera topic so the clang format shifted all the strings to their own lines. Interesting....

**Comment by RustyBamboo** [comment]
Fixed lol. My trust in clangformat has fallen.

**Comment by RustyBamboo** [comment]
How do I get rid of these ghost submodule diffs in roboteq?????

**Comment by kev-the-dev** [comment]
Ummm ammend commit and check it out to master I guess

**Pull request merged**